app/assets/javascripts/discourse/app/lib/hashtag-types/base.js

```diff
@@ -0,0 +1,17 @@
+export default class HashtagTypeBase {
+  constructor(container) {
+    this.container = container;
+  }
```

Comment:
I think it would be better to use the ember ownership system for this. That will allow to e.g. inject services into subclasses.

```suggestion
import { setOwner } from "@ember/application";

export default class HashtagTypeBase {
  constructor(container) {
    setOwner(this, container);
  }
```

```js
import HashtagTypeBase from "discourse/lib/hashtag-types/base";
import { inject as service } from "@ember/service";

export default class ChannelHashtagType extends HashtagTypeBase {
  @service currentUser;
  @service chatChannelsManager;

  get type() {
    return "channel";
  }

  get preloadedData() {
    if (this.currentUser) {
      return this.chatChannelsManager.publicMessageChannels;
    } else {
      return [];
    }
  }

  …
}
```

Reply:
Ah much better thank you :pray: Funny thing is I used this in chat recently and just...forgot :laughing: